### PR TITLE
fix: key prefix preventing jumpstart model repack

### PR DESCRIPTION
**Pull request by evakravi**
*Description of changes:*
Repack is prevented if a key prefix is supplied and it's a `FrameworkModel`. This fixes a bug where repacking was prevented for JumpStart models. 

*Testing done:*
Current unit tests: PASS
Studio Testing: PASS -- this fix resolves the problem when used in the JumpStart Studio environment. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -592,7 +592,12 @@ def prepare_container_def(
         deploy_env = copy.deepcopy(self.env)
         if self.source_dir or self.dependencies or self.entry_point or self.git_config:
             is_repack = (
-                self.source_dir and self.entry_point and not (self.key_prefix or self.git_config)
+                self.source_dir
+                and self.entry_point
+                and self.model_data
```
Comment (JGuinegagne) [inline src/sagemaker/model.py:597]: sanity-check: are you sure `self.model_data` is always required?

For example, could there be cases where the script gets passed as `file://` (from local file system) with no model S3 URI ?

**Review by JGuinegagne — CHANGES_REQUESTED** [review]
Please add unit test coverage

File: src/sagemaker/model.py (hunk 1 above)
Comment (evakravi) [reply src/sagemaker/model.py:597]: actually, good catch. i didn't intend to commit that. let me re-run the unit tests to confirm...

**Review by JGuinegagne — APPROVED** [review]

**Review by knikure — COMMENTED** [review]
/bot run slow-tests

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run notebook-tests

**Pull request merged**